packages/contracts-bedrock/src/L1/LGEMigrationV1.sol

```diff
@@ -0,0 +1,165 @@
+        address _usdxBridge,
+        address _lgeStaking,
+        address _usdc,
+        address _wstETH,
+        address[] memory _l1Addresses,
+        address[] memory _l2Addresses,
+        address[] memory _restrictedL2Addresses
+    ) {
+        _transferOwnership(_owner);
+        l1StandardBridge = IL1StandardBridge(_l1StandardBridge);
+        l1LidoTokensBridge = IL1LidoTokensBridge(_l1LidoTokensBridge);
+        usdxBridge = IUSDXBridge(_usdxBridge);
+        lgeStaking = _lgeStaking;
+        usdc = _usdc;
+        wstETH = _wstETH;
+        uint256 length = _l1Addresses.length;
+        require(
+            length == _l2Addresses.length,
+            "LGE Migration: L1 addresses array length must equal the L2 addresses array length."
+        );
+        for (uint256 i; i < length; ++i) {
+            l1ToL2Addresses[_l1Addresses[i]] = _l2Addresses[i];
+            gasLimits[_l1Addresses[i]] = 21000;
+        }
+        length = _restrictedL2Addresses.length;
+        for (uint256 j; j < length; ++j) {
+            restrictedL2Addresses[_restrictedL2Addresses[j]] = true;
+        }
+    }
+
+    /// @notice This function is called by the LGE Staking contract to facilitate migration of staked tokens from
+    ///         the LGE Staking pool to the Ozean L2.
+    /// @param _l2Destination The address which will be credited the tokens on Ozean.
+    /// @param _tokens The tokens being migrated to Ozean from the LGE Staking contract.
+    /// @param _amounts The amounts of each token to be migrated to Ozean for the _user
+    function migrate(address _l2Destination, address[] calldata _tokens, uint256[] calldata _amounts)
+        external
+        nonReentrant
+    {
+        require(msg.sender == lgeStaking, "LGE Migration: Only the staking contract can call this function.");
+        require(!restrictedL2Addresses[_l2Destination], "LGE Migration: L2 address recipient restricted.");
+        uint256 length = _tokens.length;
+        for (uint256 i; i < length; i++) {
+            require(
+                l1ToL2Addresses[_tokens[i]] != address(0), "LGE Migration: L2 address not set for migration."
+            );
+            if (_tokens[i] == usdc) {
+                /// Handle USDC
+                IERC20(_tokens[i]).safeApprove(address(usdxBridge), _amounts[i]);
+                usdxBridge.bridge(_tokens[i], _amounts[i], _l2Destination);
+            } else if (_tokens[i] == wstETH) {
+                /// Handle wstETH
+                IERC20(_tokens[i]).safeApprove(address(l1LidoTokensBridge), _amounts[i]);
+                l1LidoTokensBridge.depositERC20To(
+                    _tokens[i], l1ToL2Addresses[_tokens[i]], _l2Destination, _amounts[i], gasLimits[_tokens[i]], ""
+                );
+            } else {
+                /// Handle other tokens
+                IERC20(_tokens[i]).safeApprove(address(l1StandardBridge), _amounts[i]);
+                l1StandardBridge.depositERC20To(
```

Comment:
Does Hyperlane bridge have the same interface 'bridge.depositERC20To()` or we are going to use `l1standardBridge` for the LGE migration ?

Reply:
Hyperlane have their own contract interface, the migration contract will need to be updated given the bridging solution per asset we decide for Ozean.